### PR TITLE
Use Notify API key for service ID

### PR DESCRIPTION
**Pull request by timarney**
Update List Manager to use the Notify Service as the one and only sending service for List Manager.

Screen Shot 2022-03-16 at 10 28 29 AM

PR Includes:
- https://github.com/cds-snc/gc-articles-issues/issues/264
- https://github.com/cds-snc/gc-articles-issues/issues/265
- https://github.com/cds-snc/gc-articles-issues/issues/282

@pcraig3 --- ☝️  we can tweak these but dropped in some starter code

TODO:
- [x] List Manager auto-populate Service ID when creating / updating a list (Tim)
- [x] List Manager --- skip the `Your Services` landing page (Tim)
- [ ] Update Cypress tests (Dave)
- [x] ~Handle missing Service ID / API key `get_option('NOTIFY_API_KEY')` --- note we can `default` to the `default` --- as we need to ensure we only show the users Services. (Dave)~  Tim ended up doing a params check vs data passed in check for now.

Nice to have:

Prompt if users haven't entered an API key when visiting List Manager.

## Testing
- [x] Create a new site
- [x]  Add Notify API key -- for the sending service
- [x] API key has description (UI)
- [x] Template ID has description (UI)
- [x] Create a new list via List Manager
- [x] Can add the subscribe block
- [x] Subscribe block -> don't select a list ID --- block should show no block select + not show on the front-end
- [x] Subscribe block -> select a list... this should be the list you send to
- [x] Subscribe to the list + confirm your email
- [x] Send to the List using the updated `Bulk Send` 
- [x] Shows the List Count dialog confirm before send
- [x] Sends via your Notify sending service

### CODE REVIEW

**Comment by timarney** [comment]
Saving the lists from List Manager is currently hitting an error. 

Screen Shot 2022-03-16 at 2 14 41 PM

**Comment by timarney** [comment]
@pcraig3 

I think there's an issue with `onfirmSubscription(): array` when subscribing 

We're returning  ```return json_encode($this->subscribe($email, $listId));```

> Returns a JSON encoded string on success or false on failure.

Can you confirm?

**Comment by timarney** [comment]
@dsamojlenko @pcraig3 --- outside of the subscribe issue + I can't bulk send --- the rest of the today items seem to be functional.

**Comment by dsamojlenko** [comment]
saving this for later
```
apache        | [Wed Mar 16 20:11:03.912313 2022] [proxy_fcgi:error] [pid 10:tid 140703105051448] [client 172.20.0.1:59740] AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught TypeError: substr(): Argument #1 ($string) must be of type string, bool given in /usr/src/wordpress/wp-content/plugins/cds-base/classes/Modules/Notify/Utils.php:44\nStack trace:\n#0 /usr/src/wordpress/wp-content/plugins/cds-base/classes/Modules/Notify/Utils.php(44): substr()\n#1 /usr/src/wordpress/wp-content/plugins/cds-base/classes/Modules/Notify/ListManagerSettings.php(58): CDS\\Modules\\Notify\\Utils::extractServiceIdFromApiKey()\n#2 /usr/src/wordpress/wp-includes/class-wp-hook.php(307): CDS\\Modules\\Notify\\ListManagerSettings->listManagerAppPage()\n#3 /usr/src/wordpress/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters()\n#4 /usr/src/wordpress/wp-includes/plugin.php(474): WP_Hook->do_action()\n#5 /usr/src/wordpress/wp-admin/admin.php(259): do_action()\n#6 {main}\n  thrown in /usr/src/wordpress/wp-content/plugins/cds-base/classes/Modules/Notify/Utils.php on line 44', referer: http://localhost/wp-admin/admin.php?page=cds_notify_send
```

File: cypress/integration/list-manager.settings.spec.js [hunk 1]
```diff
@@ -7,15 +7,15 @@ describe('List Manager Settings', () => {
...
+  it.skip('Displays settings screen', () => {
     cy.visit('/wp-admin/admin.php?page=cds_list_manager_settings');
 
     cy.get('h1').should('have.text', 'List Manager Settings');
 
     cy.get('.api-key').should('be.empty');
   });
 
-  it('Can save settings', () => {
+  it.skip('Can save settings', () => {
```
Comment (timarney) [inline cypress/integration/list-manager.settings.spec.js:18]: We should be able to remove this test -- page doesn't exist

**Review by dsamojlenko — APPROVED** [review]

**Pull request merged**